### PR TITLE
Late day warning

### LINKED ISSUE #1754: Warning When Using More Late Days Than A Student Has
When a student tries to submit an assignment 3 days late when they only have 2 late days, they should be warned.

### PR DESCRIPTION
**Pull request by RylanOC**
Added a banner at the top of the submission page to show the number of late days available for an assignment.
screenshot 2018-02-25 13 44 56

When the assignment is late, the banner also displays the number of days it is late by.
screenshot 2018-02-25 13 56 13

Finally, when a student is late by more days than they have late days to make up for, the banner turns red.
screenshot 2018-02-25 13 58 09

closes #1754

### CODE REVIEW

File: site/app/views/submission/HomeworkView.php [hunk 1]
```diff
@@ -67,6 +68,32 @@ public function showGradeable($gradeable, $late_days_use, $extensions) {
 </div>
 HTML;
         }
+        $ldu = new LateDaysCalculation($this->core, $gradeable->getUser()->getId());
+        $late_days_data = $ldu->getGradeable($gradeable->getUser()->getId(), $gradeable->getId());
+        $late_days_remaining = $late_days_data['remaining_days'];
```
Comment (emaicus) [inline site/app/views/submission/HomeworkView.php]: Late days remaining is not necessarily the number of late days allowed for an assignment. This is misleading.

File: site/app/views/submission/HomeworkView.php [hunk 2]
```diff
@@ -67,6 +68,32 @@ public function showGradeable($gradeable, $late_days_use, $extensions) {
...
+        $days_late = $gradeable->getActiveDaysLate();
```
Comment (emaicus) [inline site/app/views/submission/HomeworkView.php]: This function is still using the 5 minute grace period. This leads to some inconsistencies, as late day deduction does not seem to use the grace period.

File: site/app/views/submission/HomeworkView.php [hunk 3]
```diff
@@ -67,6 +68,32 @@ public function showGradeable($gradeable, $late_days_use, $extensions) {
...
+        }
+        else {
+            if ($late_days_remaining > $days_late) {
+                $style = '';
+            }
+            else {
+                $style = 'background-color: #d9534f;';
+            }
+            $return .= <<<HTML
+<div class="content" style=$style>
```
Comment (emaicus) [inline site/app/views/submission/HomeworkView.php]: Replace $style with "{$style}"

**Comment by bmcutler** [comment]
Mostly good.  But late day status for deadlines other than midnight might be incorrect.  More debugging necessary after #1164 is in progress/completed.

**Pull request merged**